partiql-eval/src/main/kotlin/org/partiql/eval/internal/SqlCompiler.kt

```diff
@@ -348,46 +348,45 @@ internal class SqlCompiler(
         }
 
         override fun visitCallDynamic(rex: RexCallDynamic, ctx: Unit): Operator.Expr {
-            // Check candidate name and arity for uniformity
+            // Check candidate arity for uniformity
             var arity: Int = -1
-            var name = "unknown"
+            val name = rex.getName()
             // Check the candidate list size
             val functions = rex.getFunctions()
             if (functions.isEmpty()) {
-                error("Rex.Op.Call.Dynamic had an empty candidates list: $rex.")
+                error("Dynamic call had an empty candidates list: $rex.")
             }
             // Compile the candidates
             val candidates = Array(functions.size) {
                 val fn = functions[it]
-                val fnArity = fn.signature.parameters.size
-                val fnName = fn.signature.name.uppercase()
+                val fnArity = fn.parameters.size
                 if (arity == -1) {
+                    // set first
                     arity = fnArity
-                    name = fnName
                 } else {
                     if (fnArity != arity) {
                         error("Dynamic call candidate had different arity than others; found $fnArity but expected $arity")
                     }
-                    if (fnName != name) {
-                        error("Dynamic call candidate had different name than others; found $fnName but expected $name")
-                    }
                 }
+                // make a candidate
                 fn
             }
             val args = rex.getArgs().map { compile(it, ctx).catch() }.toTypedArray()
             return ExprCallDynamic(name, candidates, args)
         }
 
-        override fun visitCallStatic(rex: RexCallStatic, ctx: Unit): Operator.Expr {
-            val fn = rex.getFunction()
-            val args = rex.getArgs().map { compile(it, ctx) }
-            val fnTakesInMissing = fn.signature.parameters.any {
-                it.getType().kind == PType.Kind.DYNAMIC // TODO: Is this needed?
-            }
-            return when (fnTakesInMissing) {
-                true -> ExprCallStatic(fn, args.map { it.catch() }.toTypedArray())
-                else -> ExprCallStatic(fn, args.toTypedArray())
-            }
+        override fun visitCall(rex: RexCall, ctx: Unit): Operator.Expr {
+            val function = rex.getFunction()
+            val args = rex.getArgs()
+            // TODO HANDLE CALLS WITH DYNAMIC INPUT
+            // val fnTakesInMissing = fn.getParameters().any {
+            //     it.getType().kind == PType.Kind.DYNAMIC // TODO: Is this needed?
```

Comment:
Yes, I believe this is needed. If a function allows missing-valued arguments, then, in permissive mode, we need to be able to appropriately handle type exceptions that result in MISSING.

A scenario to consider is the equals function. In its function signature, it accepts dynamic. If run in permissive mode, we should be able to execute:
```partiql
{ 'a': 1 }.b = MISSING
```

I believe the PartiQL Spec says something along the lines of: "in permissive mode, equals never returns missing". Similar scenarios are encountered when we look at the logical operators.